### PR TITLE
interface for generic mr post processing

### PR DESCRIPTION
**Pull request by jtblack-aws**
*Issue #, if available:*

*Description of changes:*
Updating model runner to use a more generic "MRPostProcessing" list to read in feature selection/fusion options.

Sample Post-processing with NMS:

```
[
    {
        "step": "FEATURE_DISTILLATION",
        "algorithm": {
            "algorithmType": "NMS",
            "iouThreshold": 0.75
        }
    }
]
```

Sample Post-processing with Soft NMS:

```
[
    {
        "step": "FEATURE_DISTILLATION",
        "algorithm": {
            "algorithmType": "SOFT_NMS",
            "iouThreshold": 0.75,
            "skipBoxThreshold": 0.0001,
            "sigma": 0.1
        }
    }
]
```

Added unit tests for new code, passed centerpoint, flood, and aircraft integ tests.  Also used cesium globe to successfully submit image requests for small and large image with aircraft model.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/aws/osml/model_runner/common/mr_post_processing.py [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+
+
+class FeatureDistillationAlgorithmType(str, MRPostProcessingAlgorithmType):
+    """
+    Enum for defining different feature distillation algorithms used in post-processing.
+    Each member represents a specific algorithm for entity selection or fusion.
+    """
+
+    NMS = auto()
+    SOFT_NMS = auto()  # gaussian Soft-NMS (as opposed to linear)
```
Comment (drduhe) [inline src/aws/osml/model_runner/common/mr_post_processing.py:47]: Remove inline comment

File: src/aws/osml/model_runner/common/mr_post_processing.py [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+    """
+    Abstract base class representing a generic MR post-processing algorithm.
+    :param algorithm_type: The type of the algorithm, represented by a specific enum.
+    """
+
+    algorithm_type: MRPostProcessingAlgorithmType
+
+
+@dataclass(frozen=True)
+class FeatureDistillationAlgorithm(MRPostProcessingAlgorithm):
```
Comment (drduhe) [inline src/aws/osml/model_runner/common/mr_post_processing.py:71]: Pydoc comment

**Review by drduhe — CHANGES_REQUESTED** [review]

File: src/aws/osml/model_runner/common/mr_post_processing.py (hunk 1 above)
Comment (jtblack-aws) [reply src/aws/osml/model_runner/common/mr_post_processing.py]: moved comment to pydoc string

File: src/aws/osml/model_runner/common/mr_post_processing.py (hunk 2 above)
Comment (jtblack-aws) [reply src/aws/osml/model_runner/common/mr_post_processing.py:75]: added pydoc comment

**Review by drduhe — DISMISSED** [review]
LGTM

**Review by edparris — APPROVED** [review]

File: src/aws/osml/model_runner/common/mr_post_processing.py [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+    def __call__(cls, *args, **kw):
+        if getattr(cls, "__abstractmethods__", None):
+            raise TypeError(
+                f"Can't instantiate abstract class {cls.__name__} " f"with frozen methods {set(cls.__abstractmethods__)}"
+            )
+        return super().__call__(*args, **kw)
+
+    # @staticmethod
+    # def _generate_next_value_(name, start, count, last_values) -> str:
+    #     return name
```
Comment (RanbirAulakh) [inline src/aws/osml/model_runner/common/mr_post_processing.py:33]: Should remove the commented out lines

**Review by RanbirAulakh — APPROVED** [review]

File: src/aws/osml/model_runner/common/mr_post_processing.py (hunk 3 above)
Comment (jtblack-aws) [reply src/aws/osml/model_runner/common/mr_post_processing.py]: removed

**Pull request merged**